src/generic/gridsel.cpp

```diff
@@ -48,13 +48,55 @@ wxGridSelection::wxGridSelection( wxGrid * grid,
 {
     m_grid = grid;
     m_selectionMode = sel;
+    m_isSelecting = false;
 }
 
 bool wxGridSelection::IsSelection()
 {
     return !m_selection.empty();
 }
 
+
+void wxGridSelection::StartSelecting()
+{
+    m_isSelecting = true;
+}
+void wxGridSelection::EndSelecting()
+{
+    m_isSelecting = false;
+    //send the RANGE_SELECTED events
+    if (IsSelection())
+    {
+        size_t n;
+        wxRect r;
+        wxGridCellCoords coords1, coords2;
+
+        // deselect all blocks and update the screen
+        while ( ( n = m_selection.size() ) > 0)
```

Comment:
I'm not really sure why do we have to send events in this (reverse) order. Couldn't we just use a simple `for` loop here?

It's done like this in `ClearSelection()` because it modifies the selection while it iterates over it, but this is not the case here...

Reply:
This was just copy paste from what was there, for loop would be fine.

Reply:
New commit added which fixes this.